registry/storage/driver/oss/oss.go

```diff
@@ -360,6 +361,11 @@ func (d *driver) List(ctx context.Context, opath string) ([]string, error) {
 	directories := []string{}
 
 	for {
+		// This is to cover for the cases when the first key of listResponse.Contents equal to ossPath.
```

Comment:
It seem to OSS regard the key as a file rather than a directory.So the key will appear in `listResponse.Contents` on the first call because of its alphabetical order.   
I should move the filter out of the for loop.  
Thanks for your review